### PR TITLE
[DEV-1695] Add Google Cloud Storage support for spark

### PR DESCRIPTION
**Pull request by kchua78**
## Description

Add support to support GCS storage for Spark / Databricks sessions using service account.

## Related Issue

https://featurebyte.atlassian.net/browse/DEV-1695

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

**Review by bushwhackr — APPROVED** [review]

File: featurebyte/models/credential.py [hunk 1]
```diff
@@ -181,13 +182,32 @@ class S3StorageCredential(BaseStorageCredential):
...
     )
 
 
+class GCSStorageCredential(BaseStorageCredential):
+    """
+    Data class for a GCS storage credential.
+
+    Examples
+    --------
+    >>> gcs_storage_credential = S3StorageCredential(
```
Comment (jevonyeoh) [inline featurebyte/models/credential.py:196]: update class here to `GCSStorageCredential`

File: featurebyte/session/base_spark.py [hunk 2]
```diff
@@ -77,11 +82,30 @@ def _initialize_storage(self) -> None:
         if self.storage_type == StorageType.FILE:
             self._storage = FileSimpleStorage(storage_url=self.storage_url)
         elif self.storage_type == StorageType.S3:
+            if self.storage_credential is None:
```
Comment (jevonyeoh) [inline featurebyte/session/base_spark.py:85]: can consider extracting to a function for reuse with the GCS validation

eg. `def _validate_storage_credential(credential, expected_type)`

**Review by jevonyeoh — APPROVED** [review]
some small comments, lgtm otherwise!

File: featurebyte/models/credential.py [hunk 3]
```diff
@@ -181,13 +182,32 @@ class S3StorageCredential(BaseStorageCredential):
...
+    ...   service_account_info="google_api_token"
+    ... )
+    """
+
+    __fbautodoc__ = FBAutoDoc(proxy_class="featurebyte.GCSStorageCredential")
+
+    type: StorageCredentialType = Field(StorageCredentialType.GCS, const=True)
+    service_account_info: Dict[str, str] = Field(
```
[the remark this answers is not in the document]
Comment (kchua78) [reply featurebyte/models/credential.py:204]: Will update the example

File: featurebyte/session/base_spark.py (hunk 2 above)
Comment (kchua78) [reply featurebyte/session/base_spark.py:85]: Will use typecheck instead to simplify credential checking

Comment (kchua78) [reply featurebyte/session/base_spark.py:85]: If we move the type checking elsewhere mypy type checking fails. Will keep current implementation first and optimize later when adding other storage types

**Pull request merged**